app/routes/routes.py

```diff
@@ -0,0 +1,266 @@
+
+from app import app, cipher_suite, login_manager
+from app.services.image_upload import get_image
+from app.services.geolocalization_services.user_localization_and_distance import (country_name_to_code,
+                                                generate_random_ip, get_cities,
+                                                get_location, get_coordinates)
+from models import RobotProfile, User, UserRobot
+from services.database_operations.database_operations import (get_user_from_User_table,
+                                        if_user_is_admin,
+                                        insert_into_robots_db,
+                                        insert_user_and_user_profile,
+                                        select_all_from_database,
+                                        user_in_database,
+                                        update_user_location)
+from forms.forms import (AdminForm, LoginForm, RegistrationForm, UploadForm,
+                          UserCriteria, UserProfileForm)
+
+from app import bcrypt
+
+class User(UserMixin):
+    pass
+
+site_session = session
+
+@app.route("/")
+@app.route("/home")
+def home():
+    # print(get_location())
+    # print(country_name_to_code())
+    #print(get_cities())
+    # x = country_name_to_code("Poland")
+    # print(x)
+    # print(get_cities(x))
+    print("Warsaw")
+    if current_user.is_authenticated:
+        return redirect(url_for("user_homepage"))
+    else:
+        return render_template("main/start-page.html")
+
+
+@app.route("/lore")
+def lore():
+    return render_template("main/lore.html")
+
+
+@app.route("/register", methods=["GET", "POST"])
+def register():
+    form = RegistrationForm()
+    if form.validate_on_submit():
+        if not user_in_database(form.password.data, form.username.data):
+            user_password = form.password.data
+            site_session["username"] = form.username.data
+            site_session["email"] = form.email.data
+            site_session["password"] = form.password.data #cipher_suite.encrypt(user_password.encode())
+            return redirect(url_for("user_profile"))
+        else:
+            message = "Przykro mi ale masz nieprawidłowe dane"
+            return render_template(
+                "auth/register_form.html", form=form, register_failure_message=message
+            )
+    return render_template("auth/register_form.html", form=form)
+
+
+@app.route("/login", methods=["GET", "POST"])
+def login():
+    form = LoginForm()
+    if form.validate_on_submit():
+        if user_in_database(form.password.data) and if_user_is_admin(
+            form.name.data, form.password.data
+        ):
+            user = get_user_from_User_table(form.name.data)
+            login_user(user)
+            print(current_user.name)
+            return redirect(url_for("user_homepage"))
+
+        elif if_user_is_admin(form.name.data, form.password.data):
+            return redirect(url_for("admin_site"))

```

Comment:
you call `if_user_is_admin` twice

Reply:
https://github.com/rcybulski1122012/agh-bd2/blob/3ff58a9fe0a9799425751d142ca27a34ac8d42e1/library/auth/routes.py#L29

Have a look at this

Reply:
You can store `isAdmin` flag in the User model